src/pages/workspaces/workspace/analysis/modals/DeleteDiskChoices.ts

```diff
@@ -0,0 +1,199 @@
+      toolLabel:string
+      cloudService: any
+    }) => {
+  const getCurrentMountDirectory = () => {
+    const rstudioMountPoint = '/home/rstudio'
+    const jupyterMountPoint = '/home/jupyter'
+    const noMountDirectory = `${jupyterMountPoint} for Jupyter environments and ${rstudioMountPoint} for RStudio environments`
+    return toolLabel ?
+      (toolLabel === 'RStudio' ? rstudioMountPoint : jupyterMountPoint) :
+      noMountDirectory
+  }
+
+  return (
+    h(Fragment, [
+      h(RadioBlock, {
+        name: 'keep-persistent-disk',
+        labelText: 'Keep persistent disk, delete application configuration and compute profile',
+        checked: !deleteDiskSelected,
+        onChange: () => setDeleteDiskSelected(false)
+      }, [
+        p(['Please save your analysis data in the directory ',
+          code({ style: { fontWeight: 600 } }, [getCurrentMountDirectory()]), ' to ensure it’s stored on your disk.']),
+        p([
+          'Deletes your application configuration and cloud compute profile, but detaches your persistent disk and saves it for later. ',
+          'The disk will be automatically reattached the next time you create a cloud environment using the standard VM compute type.'
+        ]),
+        p({ style: { marginBottom: 0 } }, [
+          'You will continue to incur persistent disk cost at ',
+          span({ style: { fontWeight: 600 } },
+            [`${persistentDiskCostDisplay} per month.`])
+        ])
+      ]),
+      h(RadioBlock, {
+        name: 'delete-persistent-disk',
+        labelText: 'Delete everything, including persistent disk',
+        checked: deleteDiskSelected,
+        onChange: () => setDeleteDiskSelected(true),
+        style: { marginTop: '1rem' }
+      }, [
+        p([
+          'Deletes your persistent disk, which will also ', span({ style: { fontWeight: 600 } }, ['delete all files on the disk.'])
+        ]),
+        p({ style: { marginBottom: 0 } }, [
+          'Also deletes your application configuration and cloud compute profile.'
+        ])
+      ]),
+      Utils.cond(
+        [toolLabel === 'RStudio', () => h(SaveFilesHelpRStudio)],
+        [cloudService === cloudServices.GCE, () => SaveFilesHelp(false)],
+        [Utils.DEFAULT, () => h(SaveFilesHelpAzure)]
+      )
+    ])
+  )
+}
+
+
+export const DeleteEnvironment = ({
+  id,
+  runtimeConfig,
+  persistentDisk,
+  deleteDiskSelected,
+  setDeleteDiskSelected,
+  setViewMode,
+  renderActionButton,
+  hideCloseButton,
+  onDismiss,
+  toolLabel,
+  computeRegion
+}:{
+  id:string
+  runtimeConfig?:RuntimeConfig|null
+  persistentDisk?:any//TODO: retype this as PersistentDisk
+  deleteDiskSelected:boolean
+  setDeleteDiskSelected:(p1:boolean)=>void
+  setViewMode: (value: React.SetStateAction<string|undefined>) => void
+  renderActionButton: ()=> React.ReactElement<any, any>
+  hideCloseButton:boolean
+  onDismiss: React.MouseEventHandler<Element>
+  toolLabel:string
+  computeRegion?:string
+}) => {
```

Comment:
`DeleteEnvironmentProps` should be typed as an interface above this component. Also, any reason why this is in same file as `DeleteDiskChoices` ?